csharp/test/Microsoft.ML.OnnxRuntime.Tests.Android/RunAllTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿using OpenQA.Selenium.Appium;
+using OpenQA.Selenium;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.ML.OnnxRuntime.Tests.Android
+{
+    [TestFixture]
+    public class RunAllTest : BrowserStackTest
+    {
+        public AppiumElement FindAppiumElement(String xpathQuery, String text)
+        {
+            IReadOnlyCollection<AppiumElement> appiumElements = driver.FindElements(By.XPath(xpathQuery));
+
+            foreach (var element in appiumElements)
+            {
+                if (element.Text.Contains(text))
+                {
+                    return element;
+                }
+            }
+            // was unable to find given element
+            throw new Exception(String.Format("Could not find {0}: {1} on the page.", xpathQuery, text));
+        }
+
+        public AppiumElement FindAppiumElementThenClick(String xpathQuery, String text)
+        {
+            AppiumElement appiumElement = FindAppiumElement(xpathQuery, text);
+            appiumElement.Click();
+            return appiumElement;
+        }
+
+        public (int, int) GetPassFailCount()
+        {
+            int numPassed = -1;
+            int numFailed = -1;
+
+            IReadOnlyCollection<AppiumElement> labelElements = driver.FindElements(By.XPath("//android.widget.TextView"));
+
+            for (int i = 0; i < labelElements.Count; i++)
+            {
+                AppiumElement element = labelElements.ElementAt(i);
+
+                if (element.Text.Equals("✔"))
+                {
+                    i++;
```

Comment:
do we need to increment `i` in these two `if` conditions or can we just let the loop run?

Reply:
Ah, the number of tests that passed is stored as a separate text element from the "✔", so I increment to make sure that the element directly after the indicator is parsed as the number passed.